### ISSUE TITLE
Allow RESPONSE_TIMEOUT to be disabled

### ISSUE THREAD

**Issue by GabrielCappelli** [channelcat/sanic#2489]
**Is your feature request related to a problem? Please describe.**
I would like to be able to disable RESPONSE_TIMEOUT so I can have long requests running until completion.

**Describe the solution you'd like**
A common solution to disable timeouts is to set the value to 0.

**Additional context**
N/A

**Comment by ahopkins** [comment]
@Tronic and I have talked about an overhaul to timeouts. This is not going to land in the upcoming release, but is absolutely something I'm hoping to get into the end of Summer release.

**Comment by Tronic** [comment]
Zero to denote infinity is IMO a bad idea, even if many do it. I've seen NaNs used quite a bit for this on Python, but `None` could also work.

**Comment by Tronic** [comment]
I believe the timeout polling function is canceled at the end of the connection, but granted, it could be entirely avoided when there are no timeouts at all. However, the no timeout situation should also be quite rare as it makes it trivial to crash your server if you don't implement timeouts or other limits yourself or externally.

**Comment by ahopkins** [comment]
Yeah, I've thought about this in the past about how we could implement per-endpoint overrides. The difficulty is that the timers live outside of routing.
